src/group/managed_group/mod.rs

```diff
@@ -96,7 +98,41 @@ impl ManagedGroup {
         credential_bundle: &CredentialBundle,
         key_packages: &[KeyPackage],
     ) -> Result<(Vec<MLSMessage>, Welcome), ManagedGroupError> {
-        unimplemented!()
+        // Create framed add proposals from key packages
+        let mut plaintext_messages: Vec<MLSPlaintext> = key_packages
+            .iter()
+            .map(|key_package| {
+                self.group
+                    .create_add_proposal(&self.aad, &credential_bundle, key_package.clone())
```

Comment:
The way this is currently used I don't see a reason not to consume the key packages in this function. The flow I see here is that the client pulls key packages from the server and then puts them in here to add the corresponding clients. So the question is whether the key packages are need at any point after this.